falco.yaml

```diff
@@ -695,6 +695,7 @@ webserver:
   # Can be an IPV4 or IPV6 address, defaults to IPV4
   listen_address: 0.0.0.0
   k8s_healthz_endpoint: /healthz
+  metrics_enabled: false
```

Comment:
I thought that the same webserver would serve both endpoints.
This brings the following question: should we have different web server instance for the metrics part ?

Reply:
Fair, you actually asked me at the beginning of we should just use the existing webserver and I said yes as for me I was referring to existing code ... What would you suggest? Or what is better practice if you want to expose multiple ports? @FedeDP what's you take on this?

I personally never set up webservers, instead if anything I tried to break into them in the past 🙃 .

Reply:
I think we should reuse the same webserver we already use; having everything on the same ports on different endpoints is ok from my PoV, indeed it is better to remember "right, on port X there is Falco webserver".

Reply:
ok then it seems settled, we keep it as is.

Reply:
Re naming see https://github.com/falcosecurity/falco/pull/3140#issuecomment-2031651529, plz use `prometheus_metrics_enabled` and update the comments, so folks folks understand how to use this config.

Reply:
Understood, however if we support multiple standards, end users would need to have a way to choose which one. In addition, right now we don't even have a comment mentioning "Prometheus", how would you guess that it is Prometheus?

Reply:
For multiple standards, that's what my `metrics_backend` suggestion was for.

I agree that a comment in the configuration file is a good starting point.

So just to solve this point, should we:
- Use `prometheus_metrics_enabled` (which is pretty self explanatory)
- Use `metrics_enabled` with a comment stating that it's currently Prometheus that is the backend

?

Reply:
Just noticing now that I didn’t read the metrics_backend properly, apologies. Yeah that works for me, perhaps metrics_backend should be an array of strings similar to the plugins config setup.

Reply:
> I have used `metrics_enabled` as I though that, while using Prometheus currently, it would be subject to change so rather than having the name hard coded in the option name, having just "web server will also provide a metrics end point" would be simpler for admins in the long run.

I agree.


> Maybe providing a `metrics_backend` at a later stage to allow switching would be cleaner and clearer.
> * Use `metrics_enabled` with a comment stating that it's currently Prometheus that is the backend

@sgaist This works for me too :+1: 

> perhaps metrics_backend should be an array of strings similar to the plugins config setup.

@incertum We noticed that an array of strings does not play well with the `-o` command line flag. So, I'd recommend using multiple `_enabled` options so that it's easy to do `-o foo.bar_enabled=true`. 
Again, I strongly believe we have to re-audit all config options and make them consistent, but it's out of scope for this PR, so I'd leave this decision for later.

Reply:
Perhaps we might want to call the config `prometheus_enabled`?
Anyway, `metrics_enabled` works fine for me too. I'd avoid the complexity to deal with possible multiple metrics backend atm.
We can always introduce the new feat as `Incubating` to be able to play with it and be free to drop it/rename it in the future: https://github.com/falcosecurity/falco/blob/master/proposals/20231220-features-adoption-and-deprecation.md#maturation-levels

Reply:
Right @leogr re arrays configs. @FedeDP  would then just prometheus_metrics_enabled be ok? I believe it could be beneficial for this to show up when string searching for metrics. In addition, its even clearer for newbies to metrics.

Reply:
+1! Looks fine to me!

Reply:
I took a look at this. To make the config file easier to use I think it's important to highlight that you need both `metrics_enabled` and `metrics.enabled` to true in order for this to have any effect. So I would suggest `prometheus_metrics_enabled` with a comment that mentions that this only does something if `metrics.enabled` is true.

Technically I would also be ok with setting `prometheus_metrics_enabled` to `true` since metrics are disabled by default but maybe let's do it after the support is stable.